packages/program/src/contract-call-script.ts

```diff
@@ -1,131 +1,204 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { arrayify, concat } from '@ethersproject/bytes';
-import type { InputValue } from '@fuel-ts/abi-coder';
-import { U64Coder, Interface } from '@fuel-ts/abi-coder';
-import type { JsonAbiArgument } from '@fuel-ts/abi-coder/dist/json-abi';
-import type { BN } from '@fuel-ts/math';
+import { WORD_SIZE, U64Coder, B256Coder, ASSET_ID_LEN } from '@fuel-ts/abi-coder';
+import { BaseAssetId, ZeroBytes32 } from '@fuel-ts/address/configs';
+import type { AbstractAddress } from '@fuel-ts/interfaces';
 import { bn, toNumber } from '@fuel-ts/math';
-import type { TransactionResultReturnDataReceipt } from '@fuel-ts/providers';
+import type {
+  CallResult,
+  TransactionResultCallReceipt,
+  TransactionResultReturnDataReceipt,
+  TransactionResultReturnReceipt,
+} from '@fuel-ts/providers';
 import { ReceiptType } from '@fuel-ts/transactions';
-
-import contractCallScriptAbi from './multicall/static-out/multicall-abi.json';
-import contractCallScriptBin from './multicall/static-out/multicall-bin';
-import { ScriptRequest } from './script-request';
+import * as asm from '@fuels/vm-asm';
+
+import { InstructionSet } from './instruction-set';
+import type { EncodedScriptCall, ScriptResult } from './script-request';
+import {
+  decodeCallResult,
+  ScriptRequest,
+  SCRIPT_DATA_BASE_OFFSET,
+  POINTER_DATA_OFFSET,
+} from './script-request';
 import type { ContractCall } from './types';
 
-const contractCallAbiInterface = new Interface(contractCallScriptAbi);
-
-function getMaxNumberOfCalls() {
-  const input = contractCallAbiInterface.jsonAbi.functions[0].inputs[0];
-  const structScriptDataType = contractCallAbiInterface.jsonAbi.types[input.type];
-  const callsType =
-    contractCallAbiInterface.jsonAbi.types[
-      (structScriptDataType.components as readonly JsonAbiArgument[])[0].type
-    ];
-  const arrayRegEx = /\[(?<item>[\w\s\\[\]]+);\s*(?<length>[0-9]+)\]/;
-
-  const match = (arrayRegEx.exec(callsType.type) as RegExpExecArray).groups as Record<
-    string,
-    string
-  >;
-  return parseInt(match.length, 10);
+type CallOpcodeParamsOffset = {
+  callDataOffset: number;
+  gasForwardedOffset: number;
+  amountOffset: number;
+  assetIdOffset: number;
+};
+
+const DEFAULT_OPCODE_PARAMS = {
+  assetIdOffset: 0,
+  amountOffset: 0,
+  gasForwardedOffset: 0,
+  callDataOffset: 0,
+};
```

Comment:
```suggestion
const DEFAULT_OPCODE_PARAMS: CallOpcodeParamsOffset = {
  assetIdOffset: 0,
  amountOffset: 0,
  gasForwardedOffset: 0,
  callDataOffset: 0,
};
```
A small type annotation